cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -284,11 +284,11 @@ protected static boolean validFieldName(String fieldName) {
      */
     @SuppressWarnings("unchecked")
     protected static boolean indexLimitReached(Index index, Map<String, Object> existingIndexes) {
-        if (index.indexType.equalsIgnoreCase(Index.TEXT_TYPE)) {
+        if (index.indexType == IndexType.TEXT) {
             for (String name : existingIndexes.keySet()) {
                 Map<String, Object> existingIndex = (Map<String, Object>) existingIndexes.get(name);
                 String type = (String) existingIndex.get("type");
-                if (type.equalsIgnoreCase(Index.TEXT_TYPE) &&
+                if (IndexType.enumValue(type) == IndexType.TEXT &&
```

Comment:
Done in 3c5e647, I think I got all the references, so when ever you interact with indexes you should get `IndexType` instead of `String`